### PR TITLE
Nathan tina/add discount amount to confirmation email

### PR DESCRIPTION
**Pull request by NathanSouphanthong**
## Notion ticket link

[Add discount amount to payment confirmation page & confirmation email](https://www.notion.so/uwblueprintexecs/Add-discount-amount-to-payment-confirmation-page-confirmation-email-cf67b336a1e840fab829548f115a4b05?pvs=4)

## Implementation description
* Added retrieveStripeCheckoutSession to stripe.utils to grab a checkout session by the chargeId
* Included discount amount when sending confirmation email to parents

## Steps to test
** Note: you must be running the local Stripe CLI in order to test this change. (brew install stripe/stripe-cli/stripe; stripe login; stripe listen -forward-to localhost:500/campers/confirm-payment)
1. Go through the register flow to signup for a camp
2. Once redirected to the Stripe checkout page use code _TEST15_  to apply 15% discount
3. You'll be redirected back to FON 
4. The camper will be updated in the database to hasPaid = true
5. Verify in the confirmation email that discount amount applied is shown (example below)
Screen Shot 2023-07-15 at 4 47 04 PM

## What should reviewers focus on?
* After being redirected to FON, we receive an error modal saying the payment was cancelled. This bug should be handled in a separate ticket.  

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/services/implementations/emailService.ts [hunk 1]
```diff
@@ -125,6 +147,7 @@ class EmailService implements IEmailService {
         uniqueCampers.length
       } campers x ${totalCampDays} total days of camp) </li> 
         <li><b>Early drop-off and late pick-up fees:</b> $${dropoffAndPickupFees} </li>
+        <li><b>Discount Amount:</b> $${discountAmount} </li>
```
Comment (BraydenRoyston) [inline backend/typescript/services/implementations/emailService.ts:150]: `{if discountAmount > 0 <li><b>Discount Amount:</b> $${discountAmount} </li> : null}`

(we don't want to show the discount line if the client didn't apply a discount)

**Review by BraydenRoyston — CHANGES_REQUESTED** [review]

File: backend/typescript/services/implementations/emailService.ts (hunk 1 above)
Comment (thywang) [reply backend/typescript/services/implementations/emailService.ts]: Thanks for the catch! It's been fixed in the latest commit.

**Review by braydenatnimble — APPROVED** [review]
Good stuff!

**Review by BraydenRoyston — APPROVED** [review]
Good stuff!

**Pull request merged**